src/Couchbase.Lite.Tests.Shared/ReplicationTest.cs

```diff
@@ -1078,16 +1067,162 @@ public void TestExpiredNotPushed()
                 doc1.SetString("expire_me", "now");
                 Db.Save(doc1);
             }
-            
+
             Db.SetDocumentExpiration(docId, DateTimeOffset.Now);
             var config = CreateConfig(true, false, false);
             var callbackCount = 0;
-            RunReplication(config, 0, 0, documentReplicated:(status, args) => { callbackCount++; });
+            RunReplication(config, 0, 0, documentReplicated: (status, args) => { callbackCount++; });
             _otherDB.Count.Should().Be(0);
             callbackCount.Should().Be(0);
             _repl.Status.Progress.Total.Should().Be(0UL);
         }
 
+        [Fact]
+        public void TestConflictResolverRemoteWins()
+        {
+            CreateReplicationConflict();
+
+            var config = CreateConfig(false, true, false);
+
+            config.ConflictResolver = new TestConflictResolver((conflict) =>
+            {
+                return conflict.RemoteDocument;
+            });
+
+            RunReplication(config, 0, 0);
+
+            var doc = Db.GetDocument("doc1");
+            doc.GetString("name").Should().Be("Lion");
+        }
+
+        [Fact]
+        public void TestConflictResolverLocalWins()
+        {
+            CreateReplicationConflict();
+
+            var config = CreateConfig(false, true, false);
+
+            config.ConflictResolver = new TestConflictResolver((conflict) =>
+            {
+                return conflict.LocalDocument;
+            });
+
+            RunReplication(config, 0, 0);
+
+            var doc = Db.GetDocument("doc1");
+            doc.GetString("name").Should().Be("Cat");
+        }
+
+        [Fact]
+        public void TestConflictResolverNullDoc()
+        {
+            CreateReplicationConflict();
+
+            var config = CreateConfig(false, true, false);
+
+            config.ConflictResolver = new TestConflictResolver((conflict) => {
+                return null;
+            });
+
+            RunReplication(config, 0, 0);
+
+            Db.Count.ShouldBeEquivalentTo(0);
+        }
+
+        [Fact]
+        public void TestConflictResolverDeletedLocalWin()
```

Comment:
Not sure what this is testing?

Reply:
This is to test the scenario ```if (localDoc.IsDeleted)
                    resolvedDoc = localDoc;```

Reply:
Why is it needed though?  All the resolver does is return null....

Reply:
Make sure that the test is checking if the "deleted" document passed to the resolution function is `null` as it should be.